src/storage/src/sink/kafka.rs

```diff
@@ -881,17 +882,22 @@ impl KafkaSinkState {
     /// Updates the latest progress update timestamp based on the given
     /// input frontier and pending rows.
     ///
-    /// This will emit an `END` record to the progress topic if the frontier
+    /// This will emit a progress record to the progress topic if the frontier
     /// advanced and advance the maintained write frontier, which will in turn
     /// unblock compaction of timestamp bindings in sources.
     ///
-    /// *NOTE*: `END` records will only be emitted when
+    /// *NOTE*: Progress records will only be emitted when
     /// `KafkaSinkConnection.consistency` points to a progress topic. The
     /// write frontier will be advanced regardless.
     async fn maybe_emit_progress(
         &mut self,
         input_frontier: AntichainRef<'_, Timestamp>,
+        as_of: &SinkAsOf<Timestamp>,
     ) -> anyhow::Result<bool> {
+        if !PartialOrder::less_equal(&as_of.frontier.borrow(), &input_frontier) {
```

Comment:
Nonblocking: may be helpful to capture some of yesterday's reasoning here... might not be obvious from the code that this is handling the startup case.

Reply:
ahh yes, will add a comment -- thanks